src/jit/codegenarm64.cpp

```diff
@@ -3435,42 +3449,45 @@ void CodeGen::genCodeForCompare(GenTreeOp* tree)
 
     genConsumeOperands(tree);
 
-    emitAttr cmpSize = EA_ATTR(genTypeSize(op1Type));
-
-    assert(genTypeSize(op1Type) == genTypeSize(op2Type));
-
-    if (varTypeIsFloating(op1Type))
+    if ((tree->gtFlags & GTF_USE_FLAGS) == 0)
     {
-        assert(varTypeIsFloating(op2Type));
-        assert(!op1->isContained());
-        assert(op1Type == op2Type);
+        emitAttr cmpSize = EA_ATTR(genTypeSize(op1Type));
 
-        if (op2->IsIntegralConst(0))
+        assert(genTypeSize(op1Type) == genTypeSize(op2Type));
+
+        if (varTypeIsFloating(op1Type))
         {
-            emit->emitIns_R_F(INS_fcmp, cmpSize, op1->gtRegNum, 0.0);
+            assert(varTypeIsFloating(op2Type));
+            assert(!op1->isContained());
+            assert(op1Type == op2Type);
+
+            if (op2->IsIntegralConst(0))
+            {
+                emit->emitIns_R_F(INS_fcmp, cmpSize, op1->gtRegNum, 0.0);
```

Comment:
Should we also assert that op2->isContained() here.

Reply:
I can add it, but this section of changed code was previously only whitespace.

Reply:
This code is "containing" the const. If lower is out of sync it could reserve an extra register for the constant and possibly lead to sub-optimal register allocation when the codegen phase doesn't   use the "extra" register.